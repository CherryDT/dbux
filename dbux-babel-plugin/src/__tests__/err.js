import template from '@babel/template';
import { runAllSnapshotTests } from '../testing/test-util';

const codes = [
  `
function f() {
  console.log(1);
  throw new Error('errrrrror');
  console.log(2);
}

f();
`
];

const visited = new Set();

const errHandlerTemplate = template(`
console.log(err);
throw err;
`);


const plugin = function ({ types: t }) {
  return {
    visitor: {
      Function(path) {
        if (visited.has(path)) return;
        visited.add(path);

        const body = path.get('body');
        body.replaceWith(t.blockStatement([
          t.tryStatement(body.node,
            t.catchClause(
              t.identifier('err'),
              t.blockStatement(errHandlerTemplate({
              }))
            ),
            null)
        ]));
      }
    }
  };
};


runAllSnapshotTests(codes, __filename, plugin, true);
