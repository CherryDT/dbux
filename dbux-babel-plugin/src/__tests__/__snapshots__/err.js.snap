// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin err.es5.js: err.es5.js 1`] = `
"
function f() {
  console.log(1);
  throw new Error('errrrrror');
  console.log(2);
}

f();

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

function f() {
  try {
    console.log(1);
    throw new Error('errrrrror');
    console.log(2);
  } catch (err) {
    console.log(err);
    throw err;
  }
}

f();
"
`;

exports[`plugin err.esNext.js: err.esNext.js 1`] = `
"
function f() {
  console.log(1);
  throw new Error('errrrrror');
  console.log(2);
}

f();

      ↓ ↓ ↓ ↓ ↓ ↓

function f() {
  try {
    console.log(1);
    throw new Error('errrrrror');
    console.log(2);
  } catch (err) {
    console.log(err);
    throw err;
  }
}

f();
"
`;
